src/duckling/canvas/drawing/entity-drawer.service.ts

```diff
@@ -1,19 +1,21 @@
 import {Component, Injectable} from '@angular/core';
-import {Container, DisplayObject} from 'pixi.js';
+import {Container, DisplayObject, Graphics} from 'pixi.js';
 
 import {BaseAttributeService} from '../../entitysystem/base-attribute.service';
 import {AssetService} from '../../project';
+import {Vector} from '../../math/vector';
 import {Entity, EntitySystem, Attribute, AttributeKey, EntityPositionService, EntitySystemService} from '../../entitysystem';
+import {TaggedEntity} from '../../entitysystem/entity';
 import {drawMissingAsset} from '../../canvas/drawing/util';
 import {EntityLayerService} from '../../entitysystem/services/entity-layer.service';
 
 import {RenderPriorityService} from './render-priority.service';
 import {DrawnConstruct, setConstructPosition} from './drawn-construct';
 
-/**
- * Function type used to draw attributes.
- */
-export type AttributeDrawer<T> = (attribute : T, assetService? : any) => DrawnConstruct;
+export type AttributeDrawer<T> = {
+    paint?(attribute : T, position : Vector, graphics : Graphics) : void,
```

Comment:
This is weird syntax: 

`paint? : (attribute : T, position : Vector, graphics : Graphics) => void`